frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -248,6 +248,42 @@ function ReaderHighlight:onTapXPointerSavedHighlight(ges)
     end
 end
 
+function ReaderHighlight:updateHighlight(page, index, side, direction)
+    if self.ui.document.info.has_pages then -- we do this only if it's epub file
```

Comment:
You can keep that, as a security.
But the best would be to not have the buttons displayed in the ButtonDialog when self.ui.document.info.has_pages.